features/storage/nvstore/source/nvstore.cpp

```diff
@@ -888,6 +935,14 @@ int NVStore::init()
         // that we found our active area.
         _active_area = area;
         _active_area_version = versions[area];
+        _max_keys = keys[area];
```

Comment:
You need to ask here if `keys[area]` is 0, then set `_max_keys` to default. The reserved field gets the value of 0, so it is required for backward compatibility.